### PR TITLE
Bump bundler 1.12.5

### PR DESCRIPTION
**Pull request by schneems**
Allows for Ruby version operators such as

```
ruby "~> 2.3.1"
```

When you bundle with this set a Ruby version is written to the `Gemfile.lock`.

### CODE REVIEW

**Comment by schneems** [comment]
Related to #483

File: hatchet.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     "sharpstone/git_gemspec",
     "sharpstone/no_lockfile",
     "sharpstone/sqlite3_gemfile",
-    "sharpstone/nokogiri_160"
+    "sharpstone/nokogiri_160",
+    "sharpstone/bundle-ruby-version-not-in-lockfile"
```
Comment (dmathieu) [inline hatchet.json]: Have you considered adding a direwolf test as well?

**Comment by schneems** [comment]
Related https://github.com/heroku/heroku-buildpack-ruby/issues/452

**Comment by swrobel** [comment]
Please release a new version with this!

**Comment by schneems** [comment]
We are blocked on a bundler issue https://github.com/bundler/bundler/issues/4627 bundler needs to get this fixed and then we can upgrade to that version and cut a release.

**Pull request merged**